R/autosyn.R

```diff
@@ -472,3 +475,210 @@ flywire_ntplot3d <- function(x, nts=c("gaba", "acetylcholine", "glutamate",
+#'
+#' # Axon-dendrite split
+#' library(hemibrainr)
+#' neurons.flow = flow_centrality(neurons.syns,
+#'   polypre = TRUE,
+#'   mode = "centrifugal")
+#' clear3d()
+#' plot3d_split(neurons.flow, WithConnectors = TRUE,
+#' transmitter = TRUE,
+#' radius = 1000, soma = 4000)
+#' }
+#' }
+flywire_neurons_add_synapses <- function(x,
+                                         connectors = NULL,
+                                         cloudvolume.url=NULL, # fafbseg.cloudvolume.url="graphene://https://prodv1.flywire-daf.com/segmentation/1.0/fly_v31"
+                                         method=c("auto", "spine", "sqlite"),
+                                         remove_autapses = TRUE,
+                                         cleft.threshold = 0,
+                                         Verbose=TRUE,
+                                         transmitters=TRUE,
+                                         local = NULL, # "/Volumes/nnautilus/projects/JanFunke"
+                                         ...) UseMethod("flywire_neurons_add_synapses")
+
+#' @export
+#' @rdname flywire_neurons_add_synapses
+flywire_neurons_add_synapses.neuron <- function(x,
+                                                connectors = NULL,
+                                                cloudvolume.url=NULL,
+                                                method=c("auto", "spine", "sqlite"),
+                                                remove_autapses = TRUE,
+                                                cleft.threshold = 0,
+                                                Verbose=TRUE,
+                                                transmitters=TRUE,
+                                                local = NULL,
+                                                ...){
+  method = match.arg(method)
+  rootid = x$flywire.id
+  if(is.null(connectors)){
+    synapses = flywire_partners(rootid = rootid,
+                                partners = "both",
+                                roots=TRUE,
+                                details=TRUE,
+                                cloudvolume.url = cloudvolume.url,
+                                method=method,
+                                Verbose=Verbose,
+                                local = local,
+                                ...)
+    if(is.null(synapses) || nrow(synapses)==0){
+      class(x) = union(c("flywireneuron", "catmaidneuron"), class(x))
+      return(x)
+    }
+    if(remove_autapses) {
+      synapses=synapses[synapses$post_id!=synapses$pre_id,,drop=FALSE]
+    }
+    # Add synapses
+    synapses %>%
+      dplyr::filter(.data$cleft_scores >= cleft.threshold) %>%
+      dplyr::mutate(x = ifelse(rootid==.data$pre_svid, .data$pre_x, .data$post_x)) %>%
+      dplyr::mutate(y = ifelse(rootid==.data$pre_svid, .data$pre_y, .data$post_y)) %>%
+      dplyr::mutate(z = ifelse(rootid==.data$pre_svid, .data$pre_z, .data$post_z)) %>%
+      dplyr::arrange(desc(.data$offset)) %>%
```

Comment:
Why do you want this in descending order of `offset`? `offset` is the row index into the Buhmann table, so if you are going to sort by it, I would use ascending order, like the original Buhmann table. Please fix!

Reply:
Oops, i had meant to use the score/cleft score. Now I will ascend by offset.